src/app/pages/camera.cpp

```diff
@@ -291,59 +497,174 @@ QWidget *CameraPage::network_cam_selector()
     return widget;
 }
 
-void CameraPage::update_network_status(QMediaPlayer::MediaStatus media_status)
+void CameraPage::connect_network_stream()
 {
-    qInfo() << "camera status changed to: " << media_status;
-
-    switch (media_status) {
-        case QMediaPlayer::LoadingMedia:
-        case QMediaPlayer::LoadedMedia:
-        case QMediaPlayer::BufferedMedia:
-            this->status->setText("connecting...");
-            break;
-        default:
-            this->status->setText("connection failed");
-            emit disconnected();
-            break;
+    videoContainer_ = this->remote_video_widget;
+
+    DASH_LOG(info) << "[CameraPage] Creating GStreamer pipeline with "<<this->config->get_cam_network_url().toStdString();
+    std::string pipeline = "rtspsrc location="+this->config->get_cam_network_url().toStdString() + " latency=300" +
+                           " ! queue " +
+                           // decodebin does some absolute magic on selecting proper plugins
+                           // more importantly, it knows just the right settings to apply to these plugins
+                           // I can recreate the exact same pipeline as it ends up using, and it will be much slower because of the lack of setting tuning.
+                           " ! decodebin"
+                           + "";
+    init_gstreamer_pipeline(pipeline, true);
+    //emit the connected signal before we resize anything, so that videoContainer has had time to resize to the proper dimensions
+    emit connected_network();
+    if(videoContainer_ == nullptr)
+    {
+        DASH_LOG(info) << "[CameraPage] No video container, setting projection fullscreen";
+        videoWidget_->setFocus();
+        videoWidget_->setWindowFlags(Qt::WindowStaysOnTopHint | Qt::FramelessWindowHint);
+        videoWidget_->showFullScreen();
+    }
+    else
+    {
+        DASH_LOG(info) << "[CameraPage] Resizing to video container";
+        videoWidget_->resize(videoContainer_->size());
+        DASH_LOG(info) << "[CameraPage] Size: "<< videoContainer_->width() << "x" << videoContainer_->height();
+
     }
+    videoWidget_->show();
```

Comment:
might want to move this up into the `else`? since the if conditional is doing the `showFullScreen`